internal/zinc-core/src/main/scala/sbt/internal/inc/ClassFileManager.scala

```diff
@@ -22,6 +22,22 @@ import java.nio.file.{ Files, Path }
 import java.util.Optional
 import scala.collection.mutable
 
+private abstract class AbstractClassFileManager(auxiliaryFiles: Iterable[AuxiliaryClassFiles])
+    extends XClassFileManager {
+  protected def allPaths(classes: Vector[Path]): Vector[Path] = {
+    // different auxiliaryFiles can produce the same associated file
+    // we remove the duplicates by calling distinct
+    (classes ++ classes.flatMap(associatedFiles)).distinct
```

Comment:
Maybe not a big deal, I worry a bit about the efficiency of `distinct` if you have a really big project. I'm not sure this is full-proof either (could you end up with a relative and absolute path to the same file that would be considered distinct?), so I'd be OK with an alternative fix where we just change `TransactionalClassFileManager#deleteImpl` to check that the file exists before calling `move` on it, your call.

Reply:
As you suggest, I removed the distinct and moved the check that the files exists just before the call to `move`. Other implementations of `ClassFileManager` seems to be safe already.